nuxeo-core/nuxeo-core-management/src/main/java/org/nuxeo/ecm/core/management/api/ProbeManager.java

```diff
@@ -36,5 +39,34 @@
     ProbeInfo getProbeInfo(String name);
 
     ProbeInfo getProbeInfo(Class<? extends Probe> probeClass);
-
+    
+    /**
+     * List of probes evaluated for a health check
+     * @since 9.3
+     */
+    Collection<ProbeInfo> getAllContributeToHealthCheckProbeInfos();
+    
+    /**
+     * 
+     * Return the status of all the probes evaluated for a healthCheck.
+     * Run them before if the last run was more than 20s ago
+     * @since 9.3
+     */
+    HealthCheckResult getOrRunHealthCheck();
+    
+    /**
+     * 
+     * The probe is taken into account for the healthCheck
+     * @since 9.3
+     */
+    void registerProbeForHealthCheck(HealthCheckProbesDescriptor descriptor);
+    
+    /**
+     * 
+     * Return the status of the given probe
+     * Run them before if the last run was more than 20s ago
+     * 
+     * @since 9.3
+     */
+    HealthCheckResult getOrRunHealthCheckSingleProbe(String probe);
```

Comment:
`SingleProbe` is redundant as it takes a single `probe` parameter, so I'd rename to `getOrRunHealthCheck`